nixos/modules/services/networking/i2pd.nix

```diff
@@ -698,8 +781,39 @@ in
       {
         User = "i2pd";
         WorkingDirectory = homeDir;
-        Restart = "on-abort";
         ExecStart = "${cfg.package}/bin/i2pd ${i2pdFlags}";
+        ## Auto restart
+        Restart = if cfg.autoRestart then "on-failure" else "no";
+        ## Graceful shutdown
+        KillSignal = if cfg.gracefulShutdown then "SIGINT" else "SIGTERM";
+        TimeoutStopSec = if cfg.gracefulShutdown then "10m" else "30s";
```

Comment:
That's a lot and it is blocking right now so a restart of the system and presumable a rebuild switch takes 10m. Why does it take so long? I wouldn't copy this from upstream tbh.

Reply:
This behavior is featured by i2p protocol itself and is disabled here by default with warning in option's description.